src/lib/edit_script.rs

```diff
@@ -0,0 +1,330 @@
+// data (collectively the "Software"), free of charge and under any and all
+// copyright rights in the Software, and any and all patent rights owned or
+// freely licensable by each licensor hereunder covering either (i) the
+// unmodified Software as contributed to or provided by such licensor, or (ii)
+// the Larger Works (as defined below), to deal in both
+//
+// (a) the Software, and
+// (b) any piece of software and/or hardware listed in the lrgrwrks.txt file
+// if one is included with the Software (each a "Larger Work" to which the Software
+// is contributed by such licensors),
+//
+// without restriction, including without limitation the rights to copy, create
+// derivative works of, display, perform, and distribute the Software and make,
+// use, sell, offer for sale, import, export, have made, and have sold the
+// Software and the Larger Work(s), and to sublicense the foregoing rights on
+// either these or other terms.
+//
+// This license is subject to the following condition: The above copyright
+// notice and either this complete permission notice or at a minimum a reference
+// to the UPL must be included in all copies or substantial portions of the
+// Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#![warn(missing_docs)]
+
+use std::cmp::max;
+use std::collections::HashSet;
+use std::fmt::Debug;
+
+use action::{ApplyAction, Delete, EditScript, Insert, Move, Update};
+use ast::{ArenaError, NodeId};
+use matchers::{EditScriptResult, MappingStore, TemporaryMappingStore, MappingType};
+
+
+/// Given a matching between two ASTs, generate a complete edit script.
+///
+/// This trait should usually be implemented on configuration objects that
+/// define thresholds and weights for a given algorithm.
+pub trait EditScriptGenerator<T: Clone + Debug + Eq + 'static> {
+    /// Given a matching between two ASTs, generate a complete edit script.
+    fn generate_script(&self, store: &mut MappingStore<T>) -> EditScriptResult<T>;
+}
+
+#[derive(Debug, Clone, PartialEq)]
+/// The Chawathe et al. (1996) algorithm does not require any configuration.
+pub struct Chawathe96Config {}
+
+impl Default for Chawathe96Config {
+    fn default() -> Chawathe96Config {
+        Chawathe96Config {}
+    }
+}
+
+impl Chawathe96Config {
+    /// Create a new configuration object, with default values.
+    pub fn new() -> Chawathe96Config {
+        Default::default()
+    }
+
+    fn align_children<T: Clone + Debug + Eq + 'static>(&self,
```

Comment:
I think it might be clearer to put the parameters on the next line, rather than having the `<...>` squish them all the way to the RHS of the screen?

Reply:
This is a choice made by rustfmt....

Reply:
You're not doing much to improve my opinion of rustfmt ;) This is just really painful to read :/ I don't mind the splitting of the args onto multiple lines, but the level of indentation is just too much for me.